lib/node/rpc.js

```diff
@@ -1544,14 +1535,10 @@ class RPC extends RPCBase {
     // The client wants a coinbasetxn
     // instead of a coinbasevalue.
     if (coinbase) {
-      const tx = attempt.toCoinbase();
-      const input = tx.inputs[0];
-
-      // Pop off the nonces.
-      input.witness.pop();
-      input.witness.compile();
+      const tx = attempt.coinbase;
 
-      tx.refresh();
+      json.merkleroot = attempt.merkleRoot.toString('hex');
+      json.witnessroot = attempt.witnessRoot.toString('hex');
```

Comment:
This is a convenient feature - client can still modify TXs in the block before submitting?